### PR TITLE
Add a page with a description for services providing gateways

### PR DESCRIPTION
**Pull request by mmattel**
When working on describing WOPI, it came up we do not have an explanation about ocis gateways.

* Adding a new gateways description page
* Fixing links to services in the ports used page (missed to fix that before)

Language review welcomed

@wkloucek pls check if my description is correct

### CODE REVIEW

File: modules/ROOT/pages/deployment/services/gateways.adoc [hunk 1]
```diff
@@ -0,0 +1,28 @@
+= Gateways
+:toc: macro
+
+:description: There is more than one service providing gateway functionalities specific for particular tasks beside other tasks these services handle. The following explanation helps sorting things out.
```
Comment (EParzefall) [inline modules/ROOT/pages/deployment/services/gateways.adoc:4]:
```suggestion
:description: There is more than one service providing gateway functionalities specific to particular tasks beside other tasks these services handle. The following explanation helps sorting things out.
```

File: modules/ROOT/pages/deployment/services/gateways.adoc [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+// toc::[]
+
+== Introduction
+
+{description} The listing below focuses on the gateway and related functionalities of the mentioned services only.
+
+== Services Providing Gateways
+
+Proxy::
+The xref:{s-path}/proxy.adoc[proxy service] provides a general purpose HTTP API gateway and manages http related tasks like transport encryption, authentication and others. Note that the proxy service routes incoming http requests to other services.
```
Comment (EParzefall) [inline modules/ROOT/pages/deployment/services/gateways.adoc:20]:
```suggestion
The xref:{s-path}/proxy.adoc[proxy service] provides a general purpose HTTP API gateway and manages HTTP-related tasks like transport encryption, authentication and others. Note that the proxy service routes incoming HTTP requests to other services.
```

File: modules/ROOT/pages/deployment/services/gateways.adoc [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
++
+NOTE: The proxy service should be for security reasons the only service that is accessible from the outside.
```
Comment (EParzefall) [inline modules/ROOT/pages/deployment/services/gateways.adoc:22]:
```suggestion
NOTE: For security reasons, the proxy service should be the only service that is accessible from the outside.
```

File: modules/ROOT/pages/deployment/services/gateways.adoc [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+Frontend::
+The xref:{s-path}/frontend.adoc[frontend service] is a datagateway and provides a http up- and download API. It for example connects to the xref:{s-path}/storage-users.adoc[storage-users service] to manage up- and download activities based on requests via the proxy service.
```
Comment (EParzefall) [inline modules/ROOT/pages/deployment/services/gateways.adoc:25]:
```suggestion
The xref:{s-path}/frontend.adoc[frontend service] is a data gateway and provides a HTTP up- and download API. For example, it connects to the xref:{s-path}/storage-users.adoc[storage-users service] to manage up- and download activities based on requests via the proxy service.
```

**Review by EParzefall — APPROVED** [review]

**Review by phil-davis — APPROVED** [review]
LGTM

File: modules/ROOT/pages/deployment/services/gateways.adoc [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+Proxy::
+The xref:{s-path}/proxy.adoc[proxy service] provides a general purpose HTTP API gateway and manages HTTP-related tasks like transport encryption, authentication and others. Note that the proxy service routes incoming HTTP requests to other services.
++
+NOTE: For security reasons, the proxy service should be the only service that is accessible from the outside.
+
+Frontend::
+The xref:{s-path}/frontend.adoc[frontend service] is a data gateway and provides a HTTP up- and download API. For example, it connects to the xref:{s-path}/storage-users.adoc[storage-users service] to manage up- and download activities based on requests via the proxy service.
+
+Gateway::
+The xref:{s-path}/gateway.adoc[gateway service] provides a CS3 API gateway for all gRPC requests.
```
Comment (mmattel) [inline modules/ROOT/pages/deployment/services/gateways.adoc:28]:
```suggestion
The xref:{s-path}/gateway.adoc[gateway service] provides a CS3 API gateway for gRPC requests to the CS3 backend.
```

File: modules/ROOT/pages/deployment/services/gateways.adoc [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+// toc::[]
+
+== Introduction
+
+{description} The listing below focuses on the gateway and related functionalities of the mentioned services only.
+
+== Services Providing Gateways
+
+Proxy::
+The xref:{s-path}/proxy.adoc[proxy service] provides a general purpose HTTP API gateway and manages HTTP-related tasks like transport encryption, authentication and others. Note that the proxy service routes incoming HTTP requests to other services.
```
Comment (mmattel) [inline modules/ROOT/pages/deployment/services/gateways.adoc:20]:
```suggestion
The xref:{s-path}/proxy.adoc[proxy service] is a general purpose HTTP API gateway and manages HTTP-related tasks like transport encryption, authentication and others. Note that the proxy service routes incoming HTTP requests to other services.
```

File: modules/ROOT/pages/deployment/services/gateways.adoc [hunk 7]
```diff
@@ -0,0 +1,28 @@
+= Gateways
+:toc: macro
+
+:description: There is more than one service providing gateway functionalities specific to particular tasks beside other tasks these services handle. The following explanation helps sorting things out.
```
Comment (mmattel) [inline modules/ROOT/pages/deployment/services/gateways.adoc:4]:
```suggestion
:description: There is more than one service providing gateway functionalities. The following explanation helps sorting things out.
```

File: modules/ROOT/pages/deployment/services/gateways.adoc [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+== Services Providing Gateways
+
+Proxy::
+The xref:{s-path}/proxy.adoc[proxy service] provides a general purpose HTTP API gateway and manages HTTP-related tasks like transport encryption, authentication and others. Note that the proxy service routes incoming HTTP requests to other services.
++
+NOTE: For security reasons, the proxy service should be the only service that is accessible from the outside.
+
+Frontend::
+The xref:{s-path}/frontend.adoc[frontend service] is a data gateway and provides a HTTP up- and download API. For example, it connects to the xref:{s-path}/storage-users.adoc[storage-users service] to manage up- and download activities based on requests via the proxy service.
```
Comment (mmattel) [inline modules/ROOT/pages/deployment/services/gateways.adoc:25]:
```suggestion
The xref:{s-path}/frontend.adoc[frontend service] the fronend acts among other things as the data gateway for the HTTP up- and download API. For example, it connects to the xref:{s-path}/storage-users.adoc[storage-users service] to manage up- and download activities based on requests via the proxy service.
```

File: modules/ROOT/pages/deployment/services/gateways.adoc [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+== Introduction
+
+{description} The listing below focuses on the gateway and related functionalities of the mentioned services only.
+
+== Services Providing Gateways
+
+Proxy::
+The xref:{s-path}/proxy.adoc[proxy service] provides a general purpose HTTP API gateway and manages HTTP-related tasks like transport encryption, authentication and others. Note that the proxy service routes incoming HTTP requests to other services.
++
+NOTE: For security reasons, the proxy service should be the only service that is accessible from the outside.
```
Comment (mmattel) [inline modules/ROOT/pages/deployment/services/gateways.adoc:22]:
```suggestion
NOTE: All services end users currently can use, are provided by the proxy. All other APIs and services need to be protected to not be accessible by end users and are only for service-to-service communication.
```

**Comment by mmattel** [comment]
@wkloucek I have made suggestable commits based on your comments. Can you have a look on if that is correct  and commit in case / resolve the corresponding comment?

**Pull request merged**